### PR TITLE
Alerting: Add choice to external alertmanagers

### LINKED ISSUE #40295: Alerting: Support configuration for sending alerts to internal and/or external AM
Add a checkbox for enable/disable sending alerts to:
- Only external alertmanagers
- Only internal alertmanager
- Both

Waiting on #37972

### PR DESCRIPTION
**Pull request by peterholmberg**
**What this PR does / why we need it**:
This PR introduces a choice to send alerts to the internal alertmanager, to any added external alertmanagers or to both. 

**Which issue(s) this PR fixes**:

Fixes #40295

**Special notes for your reviewer**:

image

### CODE REVIEW

**Review by konrad147 — APPROVED** [review]

**Comment by konrad147** [comment]
LGTM 🚀

File: public/app/features/alerting/unified/components/admin/ExternalAlertmanagers.tsx [hunk 1]
```diff
@@ -70,16 +94,26 @@ export const ExternalAlertmanagers = () => {
     }));
   }, [setModalState]);
 
+  const onChangeAlertmanagerChoice = (alertmanagersChoice: string) => {
+    dispatch(
+      addExternalAlertmanagersAction({ alertmanagers: externalAlertManagers!.map((am) => am.url), alertmanagersChoice })
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/components/admin/ExternalAlertmanagers.tsx:99]: Can we replace this with `(externalAlertmanagers ?? [])` or perhaps we should just always return `[]` from `useExternalAmSelector()` 🤔

**Review by gillesdemey — APPROVED** [review]
LGTM; just one nit

File: public/app/features/alerting/unified/components/admin/ExternalAlertmanagers.tsx (hunk 1 above)
Comment (peterholmberg) [reply public/app/features/alerting/unified/components/admin/ExternalAlertmanagers.tsx:99]: yeah nice

File: public/app/features/alerting/unified/hooks/useExternalAmSelector.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ import { StoreState } from '../../../../types';
 const SUFFIX_REGEX = /\/api\/v[1|2]\/alerts/i;
 type AlertmanagerConfig = { url: string; status: string; actualUrl: string };
 
-export function useExternalAmSelector(): AlertmanagerConfig[] | undefined {
+export function useExternalAmSelector(): AlertmanagerConfig[] | [] {
```
Comment (gillesdemey) [inline public/app/features/alerting/unified/hooks/useExternalAmSelector.ts:7]: Not strictly necessary as `AlertmanagerConfig[]` also allows the array to be empty :)

**Review by gillesdemey — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**